M2/Macaulay2/packages/RInterface/doc.m2

```diff
@@ -2,7 +2,7 @@
 -- RInterface documentation --
 ------------------------------
 
-beginDocumentation()
+beginDocumentation(OptionalComponentsPresent => RPresent)
```

Comment:
The builds are failing right now because the `RInterface` tests are loaded before this call to `beginDocumentation()` and so the `OptionalComponentsPresent` package option doesn't exist yet, which causes a problem when calling `check`.

A band-aid fix for this specific package would be to switch the order of the two `load` lines at the bottom of `RInterface.m2`, but I wonder if `TEST` should raise an error if the option doesn't exist yet, so we know that we're calling it after `beginDocumentation()`.

Reply:
Actually, that's not quite right.  We're calling `end` since R isn't found before `beginDocumentation()`.  This package is unusual in that regard.

But regardless, there should probably be some kind of check (either in `TEST` or `check`) to make sure that the `OptionalComponentsPresent` option exists to avoid strange "key not found" error messages.

Reply:
I just pushed a fix for the RInterface-specific issue.

Reply:
Hmm, I think it shouldn't matter where `beginDocumentation()` is called. Actually, I tend to prefer defining tests _before_ calling `beginDocumentation()`, because I've ran into weird issues where the tests weren't loaded since M2 stopped at `beginDocumentation()`. I'm not sure what specific situation caused this though.

For this issue, I think I might prefer treating the absence of `OptionalComponentsPresent` as it being true. I think this would solve the failure, no?

Reply:
Yeah, you're right -- it doesn't matter since we won't be calling `check` until after the package is loaded.

Yes, your idea is much better than manually setting it for a specific package like I was doing!

Reply:
I just pushed an alternate fix based on your suggestion.  One of the builds was failing even after the RInterface-specific fix, so it must not be the only package where this was an issue.